great_tables/_formats.py

```diff
@@ -3830,6 +3832,289 @@ def _build_img_tag(uri: str, height: str | None = None, width: str | None = None
+    fill_alpha
+        The level of transparency for the icon fill can be controlled with a decimal value between
+        `0` and `1`.
+    margin_left
+        The length value for the margin that's to the left of the icon. By default, `"auto"` is
+        used for this but if space is needed on the left-hand side then a length of `"0.2em"` is
+        recommended as a starting point.
+    margin_right
+        The length value for the margin right of the icon. By default, `"auto"` is used but if
+        space is needed on the right-hand side then a length of `"0.2em"` is recommended as a
+        starting point.
+
+    Returns
+    -------
+    GT
+        The GT object is returned. This is the same object that the method is called on so that we
+        can facilitate method chaining.
+
+    Examples
+    --------
+    For this first example of generating icons with `fmt_icon()`, let's make a simple DataFrame that
+    has two columns of Font Awesome icon names. We separate multiple icons per cell with commas. By
+    default, the icons are 1 em in height; we're going to make the icons slightly larger here (so we
+    can see the fine details of them) by setting height = "4em".
+
+    ```{python}
+    import pandas as pd
+    from great_tables import GT
+
+    animals_foods_df = pd.DataFrame(
+        {
+            "animals": ["hippo", "fish,spider", "mosquito,locust,frog", "dog,cat", "kiwi-bird"],
+            "foods": ["bowl-rice", "egg,pizza-slice", "burger,lemon,cheese", "carrot,hotdog", "bacon"],
+        }
+    )
+
+    (
+        GT(animals_foods_df)
+        .fmt_icon(
+            columns=["animals", "foods"],
+            height="4em"
+        )
+        .cols_align(
+            align="center",
+            columns=["animals", "foods"]
+        )
+    )
+    ```
+
+    Let's take a few rows from the towny dataset and make it so the `csd_type` column contains
+    *Font Awesome* icon names (we want only the `"city"` and `"house-chimney"` icons here). After
+    using `fmt_icon()` to format the `csd_type` column, we get icons that are representative of the
+    two categories of municipality for this subset of data.
+
+    ```{python}
+    import polars as pl
+    from great_tables import GT
+    from great_tables.data import towny
+
+
+    # Assuming `towny` is a Polars DataFrame
+    towny_mini = (
+        pl.from_pandas(towny)
+        .select(["name", "csd_type", "population_2021"])
+        .filter(pl.col("csd_type").is_in(["city", "town"]))
+        .group_by("csd_type", maintain_order=True)
+        .agg([
+            pl.col("name").sort_by("population_2021", descending=True).head(5),
+            pl.col("population_2021").sort(descending=True).head(5)
+        ])
+        .sort("csd_type")
+        .explode(["name", "population_2021"])
+        .with_columns(
+            csd_type = pl.when(pl.col("csd_type") == "town")
+            .then(pl.lit("house-chimney"))
+            .otherwise(pl.lit("city"))
+        )
+    )
```

Comment:
Is it possible to just select a few rows of this data. E.g. `towny[[0, 3, 5], :]`. Say maybe the top 2 in each of your groups. It seems easier to get into an example with less setup (a doesn't seem critical to have a comprehensive table).